File: src/main/java/net/hectus/neobb/turn/default_game/mob/TSheep.java
package net.hectus.neobb.turn.default_game.mob;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.ColdClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.SupernaturalClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import org.bukkit.entity.Sheep;

import java.util.List;

public class TSheep extends MobTurn<Sheep> implements BuffFunction, SupernaturalClazz {
    public TSheep(NeoPlayer player) { super(player); }
    public TSheep(Sheep data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 4;
    }

    @Override
    public void apply() {
        switch (data.getColor()) {
            case PINK -> player.game.win(player);
            case LIGHT_GRAY -> new Buff.ExtraTurn().apply(player);
            case GRAY -> new Buff.Luck(Buff.BuffTarget.YOU, 25).apply(player);
            case BLACK -> new Buff.Luck(Buff.BuffTarget.OPPONENTS, -15).apply(player);
            case BROWN -> player.inventory.addRandom();
            case null, default -> {
                if (player.game.allowedClazzes().contains(ColdClazz.class))
                    new Buff.ExtraTurn().apply(player);
            }
        }
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.Luck(Buff.BuffTarget.YOU, 5));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/mob/TSheep.kt
package net.hectus.neobb.modes.turn.default_game.mob

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.ColdClazz
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.SupernaturalClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.BuffFunction
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.DyeColor
import org.bukkit.entity.Sheep

class TSheep(data: Sheep?, cord: Cord?, player: NeoPlayer?) : MobTurn<Sheep>(data, cord, player), BuffFunction, SupernaturalClazz {
    override val cost: Int = 4

    override fun apply() {
        when (data!!.color) {
            DyeColor.PINK -> player!!.game.win(player)
            DyeColor.LIGHT_GRAY -> ExtraTurn().apply(player!!)
            DyeColor.GRAY -> Luck(25).apply(player!!)
            DyeColor.BLACK -> Luck(-15, Buff.BuffTarget.OPPONENTS).apply(player!!)
            DyeColor.BROWN -> player!!.inventory.addRandom()
            else -> {
                if (player!!.game.allowed.contains(ColdClazz::class))
                    ExtraTurn().apply(player)
            }
        }
    }

    override fun buffs(): List<Buff<*>> {
        return listOf(Luck(5))
    }
}
